src/lib/duration/format.js

```diff
@@ -0,0 +1,245 @@
+                }
+                //     throw 'Moment Duration Format: unable to parse token decimal value.';
+                // TODO: what should we do instead of throwing an error?
+            }
+        } else {
+            wholeValue = Math[truncMethod](value);
+            decimalValue = value - wholeValue;
+        }
+
+        // update tokens array
+        // using this algorithm to not assume anything about the order or frequency of any tokens
+        each.call(tokens, function (token) {
+            if (token.type === momentType) {
+                extend(token, {
+                    value: value,
+                    wholeValue: wholeValue,
+                    decimalValue: decimalValue,
+                    isLeast: isLeast,
+                    isMost: isMost
+                });
+            }
+        });
+
+        // update remainder
+        remainder.subtract(wholeValue, momentType);
+    });
+
+    // build output
+
+    // the first moment token can have special handling
+    foundFirst = false;
+
+    tokens = map(tokens, function (token, index) {
+        var val, decVal;
+        if (!token.type) {
+            return token.token;  // if it is not a moment token, use the token as its own value
+        }
+
+        val = token.wholeValue.toString().replace(/^\-/, ''); // remove negative sign from the beginning
+
+        // apply token length formatting
+        // special handling for the first moment token that is not the most significant in a trimmed template
+        if (token.length > 1 && (foundFirst || token.isMost)) {
+            val = zeroFill(val, token.length);
+        }
+
+        // add decimal value if precision > 0
+        if (token.isLeast && (settings.precision > 0)) {
+            decVal = token.decimalValue.toString();
+            val += settings.decimalSeparator + decVal;
+        }
+
+        // add a negative sign if the value is negative and token is the first one
+        if (token.value < 0 && !index) {
+            val = '-' + val;
+        }
+
+        foundFirst = true;
+        return val;
+    });
+
+    return this.localeData().postformat(tokens.join(''));
+}
+
+formatDuration.defaults = {
+    // token definitions
+    escape: /\[(.+?)\]/,
+    years: /[Yy]+/,
+    months: /M+/,
+    weeks: /[Ww]+/,
+    days: /[Dd]+/,
+    hours: /[Hh]+/,
+    minutes: /m+/,
+    seconds: /s+/,
+    milliseconds: /S+/,
+    general: /.+?/,
+
+    // token type names: in order of descending magnitude
+    // can be a space-separated token name list or an array of token names
+    types: 'escape years months weeks days hours minutes seconds milliseconds general',
+
+    // FORMAT OPTIONS:
+    // trunc: default behavior rounds final token value
+    // set to `true` to truncate final token value (this was the default behavior in moment-duration-format version 1)
+    trunc: false,
```

Comment:
So this is more of a round vs floor thing. But when your context is truncation, it will be different for positive/negative numbers. We can drop this feature altogether and keep it rounding (and if somebody complains we can add flooring later on).